test/helpers/smartcards.ts

```diff
@@ -1,11 +1,16 @@
 import { act } from '@testing-library/react'
 import * as GLOBALS from '../../src/config/globals'
-import { CardAPI, Election } from '../../src/config/types'
+import { CandidateContest, CardAPI, Election } from '../../src/config/types'
 import electionSample from '../../src/data/electionSample.json'
+import utcTimestamp from '../../src/utils/utcTimestamp'
 
 const election = electionSample as Election
+const contest0 = electionSample.contests[0] as CandidateContest
+const contest1 = electionSample.contests[1] as CandidateContest
+const contest0candidate0 = contest0.candidates[0]
+const contest1candidate0 = contest1.candidates[0]
 
-export const noCard: CardAPI = {
+export const noCard: Partial<CardAPI> = {
```

Comment:
This feels like the types aren't doing enough to help you out. Looking at `CardAPI`, it appears that we want to encode the fact that if `present` is `true` then `shortValue` and maybe also `longValueExists` will be set. We can do that like so:

```diff
diff --git a/src/AppRoot.tsx b/src/AppRoot.tsx
index 5bfb795..5330e34 100644
--- a/src/AppRoot.tsx
+++ b/src/AppRoot.tsx
@@ -35,6 +35,7 @@ import {
   VxMarkPlusVxPrint,
   VxPrintOnly,
   getAppMode,
+  PresentCardAPI,
 } from './config/types'
 
 import utcTimestamp from './utils/utcTimestamp'
@@ -172,7 +173,7 @@ class AppRoot extends React.Component<RouteComponentProps, State> {
   public isVoterCardExpired = (createdAt: number): boolean =>
     utcTimestamp() >= createdAt + GLOBALS.CARD_EXPIRATION_SECONDS
 
-  public processCard = ({ longValueExists, shortValue }: CardAPI) => {
+  public processCard = ({ longValueExists, shortValue }: PresentCardAPI) => {
     const cardData: CardData = JSON.parse(shortValue)
     switch (cardData.t) {
       case 'voter': {
@@ -283,7 +284,7 @@ class AppRoot extends React.Component<RouteComponentProps, State> {
     await this.writeCard(usedVoterCardData)
 
     const updatedCard = await this.readCard()
-    const updatedShortValue: VoterCardData = updatedCard.shortValue
+    const updatedShortValue: VoterCardData = updatedCard.present
       ? JSON.parse(updatedCard.shortValue)
       : {}
 
diff --git a/src/config/types.ts b/src/config/types.ts
index 67065a4..00bf9bf 100644
--- a/src/config/types.ts
+++ b/src/config/types.ts
@@ -184,11 +184,15 @@ export interface ClerkCardData extends CardData {
   readonly t: 'clerk'
   readonly h: string
 }
-export interface CardAPI {
-  present: boolean
+export type AbsentCardAPI = {
+  present: false
+}
+export type PresentCardAPI = {
+  present: true
   shortValue: string
   longValueExists?: boolean
 }
+export type CardAPI = AbsentCardAPI | PresentCardAPI
 
 // Machine ID API
 export interface MachineIdAPI {
diff --git a/test/helpers/smartcards.ts b/test/helpers/smartcards.ts
index 2d8658d..33b1675 100644
--- a/test/helpers/smartcards.ts
+++ b/test/helpers/smartcards.ts
@@ -10,7 +10,7 @@ const contest1 = electionSample.contests[1] as CandidateContest
 const contest0candidate0 = contest0.candidates[0]
 const contest1candidate0 = contest1.candidates[0]
 
-export const noCard: Partial<CardAPI> = {
+export const noCard: CardAPI = {
   present: false,
 }
 ```

Reply:
Love it! Makes sense. Thanks for this tip.